### PR TITLE
remove interactable actor

### PR DESCRIPTION
**Pull request by TheEmidee**
Updated the interaction code to only use GFFInteractableComponent

### CODE REVIEW

File: Source/GameBaseFramework/Public/Pickupable/GBFPickupable.h [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
 
-#include "Interaction/GBFInteractableActor.h"
-
 #include <CoreMinimal.h>
+#include <GameFramework/Actor.h>
 
 #include "GBFPickupable.generated.h"
 
+class UGBFInteractableComponent;
 class AGBFInteractableActor;
```
Comment (ThibNach) [inline Source/GameBaseFramework/Public/Pickupable/GBFPickupable.h]: can be removed

File: Source/GameBaseFramework/Public/Interaction/Abilities/GBFGameplayAbility_Interact.h [hunk 2]
```diff
@@ -114,22 +115,22 @@ class GAMEBASEFRAMEWORK_API UGBFGameplayAbility_Interact : public UGBFGameplayAb
...
         }
 
         TWeakObjectPtr< AActor > Actor;
-        TScriptInterface< IGBFInteractableTarget > InteractableTarget;
+        TWeakObjectPtr< UGBFInteractableComponent > InteractableComponent;
         EGBFInteractionGroup Group;
     };
 
-    void UpdateInteractableOptions( const TArray< TScriptInterface< IGBFInteractableTarget > > & interactable_targets );
+    void UpdateInteractableOptions(const TArray<UGBFInteractableComponent *>& interactable_components);
```
Comment (ThibNach) [inline Source/GameBaseFramework/Public/Interaction/Abilities/GBFGameplayAbility_Interact.h]: clang format

File: Source/GameBaseFramework/Public/Interaction/Abilities/GBFGameplayAbility_Interact.h [hunk 3]
```diff
@@ -114,22 +115,22 @@ class GAMEBASEFRAMEWORK_API UGBFGameplayAbility_Interact : public UGBFGameplayAb
...
     void OnPressCallBack( OptionHandle interaction_option );
     void UpdateIndicators();
-    void GetTargetInfos( TArray< InteractableTargetInfos > & target_infos, const TArray< TScriptInterface< IGBFInteractableTarget > > & interactable_targets ) const;
+    void GetTargetInfos(TArray< InteractableTargetInfos > & target_infos, const TArray<UGBFInteractableComponent *>& interactable_components) const;
```
Comment (ThibNach) [inline Source/GameBaseFramework/Public/Interaction/Abilities/GBFGameplayAbility_Interact.h]: same

**Review by ThibNach — CHANGES_REQUESTED** [review]

File: Source/GameBaseFramework/Public/Pickupable/GBFPickupable.h (hunk 1 above)
Comment (TheEmidee) [reply Source/GameBaseFramework/Public/Pickupable/GBFPickupable.h]: done

File: Source/GameBaseFramework/Public/Interaction/Abilities/GBFGameplayAbility_Interact.h (hunk 3 above)
Comment (TheEmidee) [reply Source/GameBaseFramework/Public/Interaction/Abilities/GBFGameplayAbility_Interact.h]: done

**Pull request merged**